README.md

```diff
@@ -1,7 +1,52 @@
 # swift-differentiation
+
+This repository is a collection of useful extensions to Swift Differentiation, the experimental Swift language feature.
+The contents of this repository extend the current implementation of Differentiable Swift to give the user access to more differentiable methods than currently provided in the Swift standard library. It also contains some workarounds for some methods that currently are not differentiable due to missing support in the language itself.
+Some of these extensions we would like to eventually upstream into the Swift standard library and for these methods this package serves as a place to incubate and battle test these methods. 
+Other extensions are meant to be phased out as the language feature itself fills the gap that these methods currently bridge (for example `Array.update(at:with:)`).
+
+The ultimate goal of this repository is for the library portion to no longer exist since all implementations are either upstreamed to [Swift](https://github.com/swiftlang/swift) or the provided workarounds are no longer needed since there's more direct language support making them obsolete. 
+We would like this repository to become a collection of Examples and documentation on the language feature however. So that people can more easily get started using Differentiable Swift! 
```

Comment:
```suggestion
We would ultimately like this repository to become a collection of Examples and documentation on the language feature so that people can more easily get started using Differentiable Swift! 
```